CS483/CS483/Kartaclysm/StateMachine/Gameplay/StateOptionsMenu.cpp

```diff
@@ -25,38 +29,147 @@ Kartaclysm::StateOptionsMenu::~StateOptionsMenu()
 void Kartaclysm::StateOptionsMenu::Enter(const std::map<std::string, std::string>& p_mContextParameters)
 {
 	m_bSuspended = false;
+	m_iOptionSelection = 0;
 
-	// If OptionsXML is specified, load options from XML and pop to previous state
-	auto find = p_mContextParameters.find("OptionsXML");
-	if (find != p_mContextParameters.end())
-	{
-		LoadOptionsFromXml(find->second);
-		m_pStateMachine->Pop();
-		return;
-	}
-
+	// Get player controlling menu (Player 0 by default, but might change if accessed from pause menu)
+	auto find = p_mContextParameters.find("Player");
+	m_iPlayer = (find == p_mContextParameters.end() ? 0 : atoi(find->second.c_str()));
+	
 	m_pGameObjectManager = new HeatStroke::GameObjectManager();
 
 	m_pGameObjectManager->RegisterComponentFactory("GOC_OrthographicCamera", HeatStroke::ComponentOrthographicCamera::CreateComponent);
 	m_pGameObjectManager->RegisterComponentFactory("GOC_Sprite", HeatStroke::ComponentSprite::CreateComponent);
 	m_pGameObjectManager->RegisterComponentFactory("GOC_TextBox", HeatStroke::ComponentTextBox::CreateComponent);
 	m_pGameObjectManager->RegisterComponentFactory("GOC_PerspectiveCamera", HeatStroke::ComponentPerspectiveCamera::CreateComponent);
+	m_pGameObjectManager->RegisterComponentFactory("GOC_MenuSlider", ComponentMenuSlider::CreateComponent);
 
 	m_pGameObjectManager->CreateGameObject("CS483/CS483/Kartaclysm/Data/Camera/camera_overlay.xml");
+	m_pGameObjectManager->CreateGameObject("CS483/CS483/Kartaclysm/Data/Menus/OptionsMenu/options_selection.xml");
+	m_pCurrentHighlight = m_pGameObjectManager->CreateGameObject("CS483/CS483/Kartaclysm/Data/Menus/OptionsMenu/options_highlight_music.xml");
+
+	// Set slider initial values
+	if (ComponentMenuSlider* pSlider = dynamic_cast<ComponentMenuSlider*>(m_pGameObjectManager->GetGameObject("music_slider")->GetComponent("GOC_Renderable")))
+	{
+		pSlider->SetSliderValue(static_cast<int>(HeatStroke::AudioPlayer::Instance()->GetMusicVolume()));
+	}
+	else
+	{
+		assert(false && "Failed to find music slider");
+	}
+
+	if (ComponentMenuSlider* pSlider = dynamic_cast<ComponentMenuSlider*>(m_pGameObjectManager->GetGameObject("sfx_slider")->GetComponent("GOC_Renderable")))
+	{
+		pSlider->SetSliderValue(static_cast<int>(HeatStroke::AudioPlayer::Instance()->GetSoundEffectsVolume()));
+	}
+	else
+	{
+		assert(false && "Failed to find sound effects slider");
+	}
+
+	// TODO: What music plays here? Or does it continue the previous music?
```

Comment:
I can only assume it would be the music we've been using for all the other menus.